src/main/java/com/marklogic/appdeployer/command/hosts/AssignHostsToGroupsCommand.java

```diff
@@ -0,0 +1,60 @@
+	
+    public AssignHostsToGroupsCommand() {
+        setExecuteSortOrder(SortOrderConstants.ASSIGN_HOSTS_TO_GROUPS);
+        setUndoSortOrder(SortOrderConstants.UNASSIGN_HOSTS_FROM_GROUPS);
+    }
+
+	@Override
+	public void execute(CommandContext context) {
+		Map<String, String> hostGroups = context.getAppConfig().getHostGroups();
+	    AdminManager adminManager = context.getAdminManager();
+		HostManager mgr = new HostManager(context.getManageClient());
+		for (Map.Entry<String, String> entry : hostGroups.entrySet()) {
+			String hostName = entry.getKey();
+			String groupName = entry.getValue();
+			// First, ensure a Manage server exists for the group
+			ServerManager serverMgr = new ServerManager(context.getManageClient(), groupName);
+			try {
+				String serverResponse = serverMgr.getPropertiesAsJson(MANAGE_SERVER_NAME, "group-id="+groupName);
```

Comment:
I think you can do an "exists" call here instead. That will avoid throwing/logging an error, which will look suspicious to a new developer. 

I think logging at the info level is warranted in both cases - if the Manage server exists, log a message confirming it, and if it doesn't exist, log a message stating that a Manage server will be created in group (name).